compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -6463,8 +6463,8 @@ private BType checkInvocationArgs(BLangInvocation iExpr, List<BType> paramTypes,
         }
 
         BType retType = typeParamAnalyzer.getReturnTypeParams(env, bInvokableType.getReturnType());
-        if (restType != symTable.semanticError &&
-                Symbols.isFlagOn(invokableSymbol.flags, Flags.NATIVE) &&
+        if (restType != symTable.semanticError && (Symbols.isFlagOn(invokableSymbol.flags, Flags.INTERFACE)
+                || Symbols.isFlagOn(invokableSymbol.flags, Flags.NATIVE)) &&
```

Comment:
Shall we extract `` out to a variable?

```suggestion
        long flags = invokableSymbol.flags;
        if (retType != symTable.semanticError &&
                (Symbols.isFlagOn(flags, Flags.INTERFACE) || Symbols.isFlagOn(flags, Flags.NATIVE)) &&
                Symbols.isFlagOn(retType.flags, Flags.PARAMETERIZED)) {
```

Reply:
Also includes the change from `restType` to `retType`.

Reply:
regarding `restType`
https://github.com/ballerina-platform/ballerina-lang/pull/33964#discussion_r757252003